### PR TITLE
Make the Grpc provider pluggable for Oxia server

### PR DESCRIPTION
**Pull request by merlimat**
Allowing to pass client & server provider for abstracting Grpc transport in server.

### CODE REVIEW

File: common/container/container.go [hunk 1]
```diff
@@ -10,15 +10,34 @@ import (
 	"oxia/common"
 )
 
-type Container struct {
+type GrpcServer interface {
```
Comment (andrasbeni) [inline common/container/container.go:13]: At this point, I think it's worth renaming the file to e.g. `grpc_server.go`

File: standalone/standalone_rpc_server.go [hunk 2]
```diff
@@ -37,7 +36,7 @@ func NewStandaloneRpcServer(bindAddress string, advertisedPublicAddress string,
 		numShards:               numShards,
 		walFactory:              walFactory,
 		kvFactory:               kvFactory,
-		clientPool:              common.NewClientPool(),
+		replicationRpcProvider:  server.NewReplicationRpcProvider(),
```
Comment (andrasbeni) [inline standalone/standalone_rpc_server.go:39]: How about: 
```suggestion
		replicationRpcProvider:  nil,
```
I haven't tested it, but if this does not work, then I think we're doing something wrong.

Comment (merlimat) [reply standalone/standalone_rpc_server.go:39]: I'm not sure I'm following this change. If we don't initialize it, we'll be passing a nil to the `LeaderController`. It won't work, but I don't understand why would that be wrong.

Comment (andrasbeni) [reply standalone/standalone_rpc_server.go:39]: This is the standalone server, so it must not try to replicate anything.

Comment (merlimat) [reply standalone/standalone_rpc_server.go:39]: Oh, it's true. Though it will crash when calling Close()

**Pull request merged**